packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -224,17 +228,37 @@ class InjectPluginsResult{
   final bool hasChanged;
 }
 
-/// Finds Flutter plugins in the pubspec.yaml, creates platform injection
-/// registries classes and add them to the build dependencies.
-///
-/// Returns whether any Flutter plugins are added and whether they changed.
-InjectPluginsResult injectPlugins({String directory}) {
+/// Injects plugins found in `pubspec.yaml` into the platform-specific projects.
+void injectPlugins({String directory}) {
   directory ??= fs.currentDirectory.path;
+  if (fs.file(fs.path.join(directory, 'example', 'pubspec.yaml')).existsSync()) {
+    // Switch to example app if in plugin project template.
+    directory = fs.path.join(directory, 'example');
+  }
   final List<Plugin> plugins = _findPlugins(directory);
-  final bool hasPluginsChanged = _writeFlutterPluginsList(directory, plugins);
+  final bool changed = _writeFlutterPluginsList(directory, plugins);
   if (fs.isDirectorySync(fs.path.join(directory, 'android')))
-    _writeAndroidPluginRegistry(directory, plugins);
-  if (fs.isDirectorySync(fs.path.join(directory, 'ios')))
-    _writeIOSPluginRegistry(directory, plugins);
-  return new InjectPluginsResult(hasPlugin: plugins.isNotEmpty, hasChanged: hasPluginsChanged);
+    _writeAndroidPluginRegistrant(directory, plugins);
+  if (fs.isDirectorySync(fs.path.join(directory, 'ios'))) {
+    _writeIOSPluginRegistrant(directory, plugins);
+    if (plugins.isNotEmpty)
+      const CocoaPods().setupPodfile(directory);
+    if (changed)
+      _ensurePodInstallIsExecutedOnNextIOSBuild(directory);
```

Comment:
Based on your PR description, wouldn't you want to run pod install here too? 

i.e. is the desired outcome that after `flutter packages get`, the user can just open Xcode and press run?

Reply:
Can we also move this into the cocoapods.dart? To keep the encapsulation and discoverability clearer.

Reply:
The intention of this PR is that `flutter packages get` brings the project into a state where platform-specific tooling works without `flutter xxx` help.

For **Android** that means Gradle builds work. Gradle nicely does both dependency resolution of non-pub packages and actual building.

The directly corresponding **iOS** situation is that Cocoapods+Xcode work. Doing `pod install` as part of `flutter packages get` would break this symmetry. That may be just an academic point, but we seem to have gone to great lengths to avoid doing `pod install` until there is no way around it (I assume because the pods repo is painful to download). If the user just wants to run her Flutter project on Android, `pod install` is not needed. We don't know the user's intention at `flutter packages get` time.

Reply:
Move into `cocoapods.dart`: good point, done.

Reply:
SG